python/python/tests/tests_server_modules/test_ft.py

```diff
@@ -198,3 +199,105 @@ async def _create_test_index_with_vector_field(
+        )
+        resultString = cast(bytes, result).decode(encoding="utf-8")
+        assert "price" in resultString and "0" in resultString and "10" in resultString
+
+        # FT.EXPLAIN on a search query that returns all data.
+        result = await ft.explain(glide_client, indexName=indexName, query="*")
+        resultString = cast(bytes, result).decode(encoding="utf-8")
+        assert "*" in resultString
+
+        assert await ft.dropindex(glide_client, indexName=indexName)
+
+        # FT.EXPLAIN on a missing index throws an error.
+        with pytest.raises(RequestError):
+            await ft.explain(glide_client, str(uuid.uuid4()), "*")
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_ft_explaincli(self, glide_client: GlideClusterClient):
+        indexName = str(uuid.uuid4())
+        await TestFt._create_test_index_for_ft_explain_commands(
+            self=self, glide_client=glide_client, index_name=indexName
+        )
+
+        # FT.EXPLAINCLI on a search query containing numeric field.
+        query = "@price:[0 10]"
+        result = await ft.explaincli(glide_client, indexName=indexName, query=query)
+        resultStringArr = []
+        for i in result:
+            resultStringArr.append(cast(bytes, i).decode(encoding="utf-8").strip())
+        assert (
+            "price" in resultStringArr
+            and "0" in resultStringArr
+            and "10" in resultStringArr
+        )
+
+        # FT.EXPLAINCLI on a search query containing numeric field and having bytes type input to the command.
+        result = await ft.explaincli(
+            glide_client, indexName=indexName.encode(), query=query.encode()
+        )
+        resultStringArr = []
+        for i in result:
+            resultStringArr.append(cast(bytes, i).decode(encoding="utf-8").strip())
+        assert (
+            "price" in resultStringArr
+            and "0" in resultStringArr
+            and "10" in resultStringArr
+        )
+
+        # FT.EXPLAINCLI on a search query that returns all data.
+        result = await ft.explaincli(glide_client, indexName=indexName, query="*")
+        resultStringArr = []
+        for i in result:
+            resultStringArr.append(cast(bytes, i).decode(encoding="utf-8").strip())
+        assert "*" in resultStringArr
+
+        assert await ft.dropindex(glide_client, indexName=indexName)
+
+        # FT.EXPLAINCLI on a missing index throws an error.
+        with pytest.raises(RequestError):
+            await ft.explaincli(glide_client, str(uuid.uuid4()), "*")
+
+    async def _create_test_index_for_ft_explain_commands(
+        self, glide_client: GlideClusterClient, index_name: TEncodable
+    ):
+        # Helper function used for creating an index having hash data type, one text field and one numeric field.
+        fields: List[Field] = []
```

Comment:
nit: this can be written in one line
```suggestion
        fields: List[Field] = [NumericField("price"), TextField("title")]
```

Reply:
I have noted this. Will fix with a followup PR, when I merge the test files.